gems/sorbet-runtime/test/types/props/struct.rb

```diff
@@ -101,6 +106,12 @@ class StructWithReqiredField < T::Struct
 
       assert_equal(true, T::Props::Utils.optional_prop?(TestStruct.props[:foo5]))
       assert_nil(TestStruct.props[:foo5][:optional])
+
+      assert_equal(true, T::Props::Utils.required_prop?(StructWithUntypedField.props[:foo0]))
```

Comment:
Added a test here to demonstrate that `T.untyped` is considered "required" which feels right to me.